File: 1313-decompress-run-length-encoded-list/1313-decompress-run-length-encoded-list.c
#include <stdlib.h>

int* decompressRLElist(int* nums, int numsSize, int* returnSize) {
    int total = 0;

    for (int i = 0; i < numsSize; i += 2) {
        total += nums[i];
    }

    int* result = (int*)malloc(sizeof(int) * total);
    *returnSize = total;

    int index = 0;

    for (int i = 0; i < numsSize; i += 2) {
        int freq = nums[i];
        int val = nums[i + 1];

        for (int j = 0; j < freq; j++) {
            result[index++] = val;
        }
    }

    return result;
}

File: 1313-decompress-run-length-encoded-list/1313-decompress-run-length-encoded-list.rs
impl Solution {
    pub fn decompress_rl_elist(nums: Vec<i32>) -> Vec<i32> {
          let mut result = Vec::new();
    for i in (0..nums.len()).step_by(2) {
        for j in 0..nums[i] {
            result.push(nums[i + 1]);
        }
    }
    result
    }
}